homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -572,9 +573,11 @@ async def async_play_media(
 
             media_id = async_process_play_media_url(self.hass, sourced_media.url)
 
-            # Remove playlist extension as it is unsupported.
+            # Exit if the source uses unsupported file.
             if media_id.endswith(".m3u"):
-                media_id = media_id.replace(".m3u", "")
+                raise BrowseError(
```

Comment:
This is not a browser, but the play media service; which should throw a HomeAssistantError (don't forget to add translations for that error).

Reply:
Ok, that should be it.